### PR TITLE
done ticket 6

### PR DESCRIPTION
**Pull request by ThePotato97**

### CODE REVIEW

File: api/__tests__/app.test.js [hunk 1]
```diff
@@ -81,7 +81,33 @@ describe("/api", () => {
...
     });
   });
-
+  describe("GET /api/articles/:article_id/comments", () => {
+    it("should return the specified article comments", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then((res) => {
+          const { comments } = res.body;
```
Comment (pippaamy) [inline api/__tests__/app.test.js:90]: check the length before the forEach

File: api/__tests__/app.test.js [hunk 2]
```diff
@@ -81,7 +81,33 @@ describe("/api", () => {
...
+          comments.forEach((comment) => {
+            expect(comment).toHaveProperty("comment_id");
+            expect(comment).toHaveProperty("votes");
+            expect(comment).toHaveProperty("created_at");
+            expect(comment).toHaveProperty("author");
+            expect(comment).toHaveProperty("body");
+            expect(comment).toHaveProperty("article_id");
+          });
+        });
```
Comment (pippaamy) [inline api/__tests__/app.test.js:103]: we need a test for an article id that could exist but doesn't

File: api/__tests__/app.test.js [hunk 3]
```diff
@@ -81,7 +81,33 @@ describe("/api", () => {
...
+    });
```
Comment (pippaamy) [inline api/__tests__/app.test.js:104]: we need another test for 200, valid ID but no comments

File: api/__tests__/app.test.js [hunk 4]
```diff
@@ -81,7 +81,33 @@ describe("/api", () => {
...
+        .expect(200)
+        .then((res) => {
+          const { comments } = res.body;
+          comments.forEach((comment) => {
+            expect(comment).toHaveProperty("comment_id");
+            expect(comment).toHaveProperty("votes");
+            expect(comment).toHaveProperty("created_at");
+            expect(comment).toHaveProperty("author");
+            expect(comment).toHaveProperty("body");
+            expect(comment).toHaveProperty("article_id");
```
Comment (pippaamy) [inline api/__tests__/app.test.js]: you can check the article ID is 1 here

File: api/controllers/articles.controller.js [hunk 5]
```diff
@@ -19,6 +20,15 @@ exports.getArticleById = (req, res, next) => {
     });
 };
 
+exports.getCommentsByArticleId = (req, res, next) => {
+  const { article_id } = req.params;
+  selectCommentsByArticleId(article_id)
+    .then((comments) => {
```
Comment (pippaamy) [inline api/controllers/articles.controller.js:26]: looks great!

**Review by pippaamy — COMMENTED** [review]
great job, you need to add a few more tests! could you pop those in and then another pr :)

File: api/controllers/articles.controller.js [hunk 6]
```diff
@@ -19,6 +20,23 @@ exports.getArticleById = (req, res, next) => {
     });
 };
 
+exports.getCommentsByArticleId = (req, res, next) => {
+  const { article_id } = req.params;
+  selectArticleById(article_id)
+    .then((article) => {
+      const firstArticle = article[0];
+      if (!firstArticle) {
```
Comment (pippaamy) [inline api/controllers/articles.controller.js]: we don't want this here! we want this error to be caught

**Review by pippaamy — COMMENTED** [review]
can we move this error handling and pop in another pr :)

File: api/controllers/articles.controller.js [hunk 7]
```diff
@@ -1,6 +1,7 @@
 const {
   selectArticleById,
   selectArticles,
+  selectCommentsByArticleId,
 } = require("../models/articles.model");
 
 exports.getArticleById = (req, res, next) => {
```
Comment (ThePotato97) [inline api/controllers/articles.controller.js:7]: shouldn't this also have error handling in the model then?

File: api/__tests__/app.test.js [hunk 8]
```diff
@@ -81,16 +81,65 @@ describe("/api", () => {
...
+    it("should return the specified article comments", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then((res) => {
+          const { comments } = res.body;
+          expect(comments).toBeInstanceOf(Array);
+          expect(comments).toHaveLength(11);
+          comments.forEach((comment) => {
+            expect(comment).toHaveProperty("comment_id");
```
Comment (JVal-NC) [inline api/__tests__/app.test.js]: I would suggest just checking the property exist is insufficient, can we also check the data type of each key's value

**Review by JVal-NC — COMMENTED** [review]
Looks good, some comments on the way you've tested your responses properties being the only comment.

**Review by pippaamy — APPROVED** [review]
nice one Chris!!

**Pull request merged**